.github/workflows/ci.yaml

```diff
@@ -7,12 +7,68 @@ on:
     branches: [main]
 
 jobs:
-  ci:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: extractions/setup-just@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: 'pip'
+      - name: Install dependencies
+        run: just install
+      - name: Run checks
+        run: just check
+  ruff-versions:
+    name: Generate test versions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: |
+          sudo apt-get install -y ripgrep
+      - name: Determine test versions
+        id: set-versions
+        run: |
+          # Get the latest release version from GitHub
+          LATEST=$( \
+            curl -L \
+              -H "Accept: application/vnd.github+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              https://api.github.com/repos/astral-sh/ruff/releases/latest \
+              | jq '.tag_name' --raw-output \
+              | cut -c2- \
+          )
+          # Get the oldest supported version from the pyproject.toml
+          OLDEST=$(rg -No '"ruff>=(.*)"' -r '$1' pyproject.toml)
```

Comment:
It's way more painful to extract the first matching group like this without ripgrep, afaict

Reply:
I would honestly be fine hard-coding this _and_ the latest version, since we already search-and-replace the latest version every time we upgrade the LSP. Fewer dependencies, easier to maintain (IMO). But of course I defer to you.

Reply:
Maybe it's just me but a sed invocation like that is entirely incomprehensible haha :)

I'd like to try this out since I wrote it up and haven't tried generating matrix test versions like this before, but at the first sign of trouble I'm happy to just switch to hard-coded versions.